packages/abstract-signer/src.ts/index.ts

```diff
@@ -117,12 +127,48 @@ export abstract class Signer {
         return Promise.resolve("");
     }
 
-    // Populates all fields in a transaction, signs it and sends it to the network
+    /**
+     * Composes a transaction which is signed and sent to the provider's network.
+     * @param transaction - the actual tx
+     */
     async sendTransaction(transaction: Deferrable<TransactionRequest>): Promise<TransactionResponse> {
-        this._checkProvider("sendTransaction");
-        const tx = await this.populateTransaction(transaction);
-        const signedTx = await this.signTransaction(tx);
-        return await this.provider.sendTransaction(signedTx);
+        const tx = await resolveProperties(transaction);
+        if (tx.to) {
+            const signed = await this.signTransaction(tx);
+            return await this.provider.sendTransaction(signed);
+        } else {
+            const contractByteCode = tx.data;
+            let chunks = splitInChunks(Buffer.from(contractByteCode).toString(), 4096);
+            const fileCreate = {
+                gasLimit: tx.gasLimit,
+                customData: {
+                    fileChunk: chunks[0],
+                    fileKey: HederaPubKey.fromString(this._signingKey().compressedPublicKey)
+                }
+            };
+            const signedFileCreate = await this.signTransaction(fileCreate);
+            const resp =  await this.provider.sendTransaction(signedFileCreate);
+            for (let chunk of chunks.slice(1)) {
+                const fileAppend = {
+                    gasLimit: tx.gasLimit,
```

Comment:
We should not enfroce gas limits on File Create and File Append tx. We should modify our checks so that `gasLimit` can be `null` in the case for `FileCreate/Append`